src/Component.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\twig_fractal\Component.
+ */
+
+namespace Drupal\twig_fractal;
+
+use Drupal\Component\Utility\Html;
+use Symfony\Component\Yaml\Yaml;
+
+class Component {
+
+  /**
+   * The pathname of the component to render.
+   *
+   * @var string
+   */
+  protected $pathname;
+
+  /**
+   * The name of the component to render.
+   *
+   * @var string
+   */
+  protected $name;
+
+  /**
+   * The names of the variants to render.
+   *
+   * @var array
+   */
+  protected $variants = [];
+
+  public function __construct($compound) {
+    list($this->pathname, $this->name, $this->variants) = $this->extractParts($compound);
+  }
+
+  public function __get($name) {
+    return $this->$name;
```

Comment:
Declaring the properties as public would be an option, if modifying their values from the outside is deemed to be okay.  If we only expect them to be read but not changed, then the properties should stay protected/private.

Explicit getter methods are more appropriate here, because the properties (1) are known and (2) the amount is very limited and (3) their declaration (including PHPDoc) exposes them in docs and IDEs, whereas magic getters are a black box.

In cases where the getter/setter methods are not really relevant / just data accessors and not doing something special, you can define such property methods at the very end of the class to keep the important stuff at the beginning.